http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/AwsCrtConfigurationUtils.java

```diff
@@ -0,0 +1,113 @@
+
+import java.time.Duration;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.crt.http.HttpMonitoringOptions;
+import software.amazon.awssdk.crt.http.HttpProxyOptions;
+import software.amazon.awssdk.crt.io.SocketOptions;
+import software.amazon.awssdk.crt.io.TlsCipherPreference;
+import software.amazon.awssdk.crt.io.TlsContext;
+import software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient;
+import software.amazon.awssdk.http.crt.ConnectionHealthConfiguration;
+import software.amazon.awssdk.http.crt.ProxyConfiguration;
+import software.amazon.awssdk.http.crt.TcpKeepAliveConfiguration;
+import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.NumericUtils;
+
+@SdkInternalApi
+public final class AwsCrtConfigurationUtils {
+    private static final Logger log = Logger.loggerFor(AwsCrtAsyncHttpClient.class);
+
+    private AwsCrtConfigurationUtils() {
+    }
+
+    public static SocketOptions buildSocketOptions(TcpKeepAliveConfiguration tcpKeepAliveConfiguration,
+                                                   Duration connectionTimeout) {
+        SocketOptions clientSocketOptions = new SocketOptions();
+
+        if (connectionTimeout != null) {
+            clientSocketOptions.connectTimeoutMs = NumericUtils.saturatedCast(connectionTimeout.toMillis());
+        }
+
+        if (tcpKeepAliveConfiguration != null) {
+            clientSocketOptions.keepAliveIntervalSecs =
+                NumericUtils.saturatedCast(tcpKeepAliveConfiguration.keepAliveInterval().getSeconds());
+            clientSocketOptions.keepAliveTimeoutSecs =
+                NumericUtils.saturatedCast(tcpKeepAliveConfiguration.keepAliveTimeout().getSeconds());
+
+        }
+
+        return clientSocketOptions;
+    }
+
+    public static HttpProxyOptions buildProxyOptions(ProxyConfiguration proxyConfiguration, TlsContext tlsContext) {
+        if (proxyConfiguration == null) {
+            return null;
+        }
+
+        HttpProxyOptions clientProxyOptions = new HttpProxyOptions();
+
+        clientProxyOptions.setHost(proxyConfiguration.host());
+        clientProxyOptions.setPort(proxyConfiguration.port());
+
+        if ("https".equalsIgnoreCase(proxyConfiguration.scheme())) {
+            clientProxyOptions.setTlsContext(tlsContext);
+        }
+
+        if (proxyConfiguration.username() != null && proxyConfiguration.password() != null) {
+            clientProxyOptions.setAuthorizationUsername(proxyConfiguration.username());
+            clientProxyOptions.setAuthorizationPassword(proxyConfiguration.password());
+            clientProxyOptions.setAuthorizationType(HttpProxyOptions.HttpProxyAuthorizationType.Basic);
+        } else {
+            clientProxyOptions.setAuthorizationType(HttpProxyOptions.HttpProxyAuthorizationType.None);
+        }
+
+        return clientProxyOptions;
+    }
+
+    public static HttpMonitoringOptions revolveHttpMonitoringOptions(ConnectionHealthConfiguration config) {
```

Comment:
Nit: there's an old misspelling here, `revolve` should be `resolve`. I think.

Reply:
The person(me) who wrote this clearly can't spell!